lib/ecto/changeset.ex

```diff
@@ -3000,6 +3001,33 @@ defmodule Ecto.Changeset do
       _ -> acc
     end
   end
+
+  @doc ~S"""
+  Traverses changeset validations and applies the given function to validations.
+
+  This behaves the same as `traverse_errors/2`, but operates on changeset
+  validations instead of errors.
+
+  ## Examples
+
+      iex> traverse_validations(changeset, &(&1))
+      %{title: [format: ~r/pattern/, length: [min: 1, max: 20]]}
+
+      iex> traverse_validations(changeset, fn
+      ...>   {:length, opts} -> {:length, "#{Keyword.get(opts, :min, 0)}-#{Keyword.get(opts, :max, 32)}"}
+      ...>   {:format, %Regex{source: source}} -> {:format, "/#{source}/"}
+      ...>   {other, opts} -> {other, inspect(opts)}
+      ...> end)
+      %{title: [format: "/pattern/", length: "1-20"]}
+  """
+  @spec traverse_validations(t, (error -> String.t) | (Changeset.t, atom, error -> String.t)) :: %{atom => [String.t | map]}
```

Comment:
```suggestion
  @spec traverse_validations(t, (error -> String.t) | (Changeset.t, atom, error -> String.t)) :: %{atom => term}
```

Reply:
I think we need to do the same change for traverse_errors... theoretically it can return anything, no?

Reply:
Don't the values still have to be a list? ie `%{atom => [term]}`

Reply:
Yes, sorry!

Reply:
👍 updated